### ISSUE TITLE
Editor does not handle backslash at line end of comments [imported]

### ISSUE THREAD

**Issue by cmaglie** [arduino/Arduino#956]
This is [Issue 956](http://code.google.com/p/arduino/issues/detail?id=956) moved from a Google Code project.
Added by 2012-06-11T10:01:57.000Z by [meiklehm...@gmail.com](http://code.google.com/u/109795282464955675286/).
Please review that bug for more context and additional comments, but update this bug.

Original labels: Type-Defect, Priority-Medium, Component-Core

### Original description

I'm using Arduino 1.0.1 and added some comments to a .h file to describe some constants. Some lines contained a backslash at the end. I was surprised that my software did not work correctly. It took me a long time to find out that the trailing backslash was the reason because in the case the next line also will be treated as comment. It would have been much easier for me to find this out if the syntax coloring would have shown the next line in "comment color", too.

Example:
unsigned char values[] = {
  45, 12, 56, // \
  45, 13, 57,
  45, 13, 58
};

will be treated as

unsigned char values[] = {
  45, 12, 56,
  45, 13, 58
};

by the compiler.

**Comment by Chris--A** [comment]
@ffissore 

In the latest build this does not appear to be the case; the sketch below compiles fine, however its an error. It should cause a compile error and fail.

```arduino

unsigned char values[] = {
45, 12, 56, // \
45, 13, 57,
45, 13, 58
};

unsigned char values1[] = {
45, 12, 56,
45, 13, 57,
45, 13, 58
};

bool test[ sizeof( values ) == sizeof( values1 ) ? 1 : 0x7FFFFFFFFFFFFFFFULL ];

void setup() {}
void loop() {}
```


On looking at the pre-processed file, it appears Arduino has removed the comment, and as a result it has un-commented the next line.

So I believe this is now a bug in the IDE pre-processor.

```arduino
#line 1 "sketch_may27a.ino"

#include "Arduino.h"
void setup();
void loop();
#line 2
unsigned char values[] = {
45, 12, 56,       
45, 13, 57,
45, 13, 58
};

unsigned char values1[] = {
45, 12, 56,
45, 13, 57,
45, 13, 58
};

bool test[ sizeof( values ) == sizeof( values1 ) ? 1 : 0x7FFFFFFFFFFFFFFFULL ];

void setup() {}
void loop() {}
```

**Comment by ffissore** [comment]
Yet another preprocessor issue

**Comment by ffissore** [comment]
Thanks for the update @Chris--A

**Comment by Chris--A** [comment]
I've proposed a fix in #3240

**Comment by ffissore** [comment]
Fixed by #3779
